src/mixins/validation-mixin.ts

```diff
@@ -0,0 +1,90 @@
+
+/**
+ * Mixin that provides some useful validation utilities.
+ */
+@Component({})
+export default class ValidationMixin extends Vue {
+  /**
+   * Creates a Vuetify rules object from the Vuelidate state.
+   * @param model The name of the model we are validating.
+   * @returns A Vuetify rules object.
+   */
+  public createVuetifyRulesObject (model: string): { [attr: string]: Array<Function> } {
+    const obj = {
+      streetAddress: [],
+      streetAddressAdditional: [],
+      addressCity: [],
+      addressRegion: [],
+      postalCode: [],
+      addressCountry: [],
+      deliveryInstructions: []
+    }
+
+    // ensure Vuelidate state object is initialized
+    if (this.$v && this.$v[model]) {
+      // iterate over Vuelidate object properties
+      Object.keys(this.$v[model])
+        // only look at validation properties
+        .filter(prop => prop.charAt(0) !== '$')
+        .forEach(prop => {
+          // create array for each validation property
+          obj[prop] = []
+          // iterate over validation property params
+          Object.keys(this.$v[model][prop].$params)
+            .forEach(param => {
+              // add specified validation functions to array
+              switch (param) {
+                case 'required': obj[prop].push(() => this.requiredRule(model, prop)); break
+                case 'minLength': obj[prop].push(() => this.minLengthRule(model, prop)); break
+                case 'maxLength': obj[prop].push(() => this.maxLengthRule(model, prop)); break
+                case 'isCanada': obj[prop].push(() => this.isCanadaRule(model, prop)); break
+                case 'isBC': obj[prop].push(() => this.isBCRule(model, prop)); break
+                // FUTURE: add extra validation functions here
+                default: break
+              }
+            })
+        })
+    }
+
+    // sample return object
+    // streetAddress: [
+    //   () => this.requiredRule('addressLocal', 'streetAddress'),
+    //   () => this.minLengthRule('addressLocal', 'streetAddress'),
+    //   () => this.maxLengthRule('addressLocal', 'streetAddress')
+    // ],
+    // ...
+
+    return obj
+  }
+
+  /**
+   * Misc Vuetify rules.
+   * @param prop The name of the property object to validate.
+   * @param key The name of the property key (field) to validate.
+   * @returns True if the rule passes, otherwise an error string.
+   */
+  protected requiredRule (prop: string, key: string): boolean | string {
+    return Boolean(this.$v[prop] && this.$v[prop][key].required) || 'This field is required'
+  }
+
+  protected minLengthRule (prop: string, key: string): boolean | string {
+    const min = this.$v[prop][key].$params.minLength.min
+    return Boolean(this.$v[prop] && this.$v[prop][key].minLength) || `Minimum length is ${min}`
+  }
+
+  protected maxLengthRule (prop: string, key: string): boolean | string {
+    const max = this.$v[prop][key].$params.maxLength.max
```

Comment:
@seeker25 Travis I'm working on migrating Create UI to Vite. I'm getting an error in `BaseAddress.vue` since the component uses this mixin:
![validation mixin error](https://github.com/bcgov/bcrs-shared-components/assets/122301442/bbc8c346-7e3f-4b06-aca1-6253eb0ab4c2)

Any way we can fix this as part of this PR?

Reply:
From the error, I'm guessing `maxLength` is null. Does it need to be declared somewhere -- in your code or in the unit test?

Reply:
Wait, I think you're right. I'll have a look.

Reply:
Unless something comes up, this should be OK. Resolving this conversation.

Reply:
Can we add optional chaining to maxLength here and to minLength on line 73?

`const min = this.$v[prop][key].$params.minLength?.min`
`const max = this.$v[prop][key].$params.maxLength?.max`

I've tried doing the above on my side and it resolved a bunch of errors.
cc: @seeker25 @severinbeauvais

Reply:
But why are those ever null/undefined? Don't we have fallback values for these or are they optional? And it they're optional, why is the code trying to get the min and max value?

(I don't want to hide the issue with conditional chaining if possible.)

Reply:
That's the thing. This hasn't been an issue before going to Vite. I'm looking at the code and I don't see the values ever being null or undefined.

Reply:
```
  protected maxLengthRule (prop: string, key: string): boolean | string {
    const max = this.$v[prop][key].$params.maxLength.max
    return Boolean(this.$v[prop] && this.$v[prop][key].maxLength) || `Maximum length is ${max}`
  }
```

So, `$v` is the Vue instance, and Vuelidate has added (or in this case, failed to add?) the right config to the prop/key object. I think those are injected by the following method (which comes from 'vue-plugin-helper-decorator'). Does that decorator work in the latest versions of Vue/Vuetify/vite?

```
export default class BaseAddress extends Mixins(ValidationMixin, CountriesProvincesMixin) {
  /**
   * The validation object used by Vuelidate to compute address model validity.
   * @returns the Vuelidate validations object
   */
  @Validation()
  public validations (): any {
    return { addressLocal: { ...this.schemaLocal } }
  }
```

Are we using the same version of Vuelidate, etc?

Reply:
> This hasn't been an issue before going to Vite.

Is changing to Vite the only change or are you also using the shared BaseAddress component instead of the common one?

Reply:
The one that's being used is from the sbc-common-components.

Regarding Vuelidate, I'm guessing that's the issue. The version of Vuelidate in Create UI is different from the ones the component is using. I'm going to give this a shot.

Reply:
Yes Sev, agreed. Resolving this conversation.